src/Microsoft.Restier.WebApi/Formatter/Deserialization/DefaultRestierDeserializerProvider.cs

```diff
@@ -11,11 +11,24 @@
 namespace Microsoft.Restier.WebApi.Formatter.Deserialization
 {
     /// <summary>
-    /// The default serializer provider.
+    /// The default deserializer provider.
     /// </summary>
-    internal class DefaultRestierDeserializerProvider : DefaultODataDeserializerProvider
+    public class DefaultRestierDeserializerProvider : DefaultODataDeserializerProvider
     {
         private RestierEnumDeserializer enumDeserializer;
+        private static readonly Lazy<DefaultRestierDeserializerProvider> lazy
```

Comment:
As you said Lewis, static instance has synchronization guarantee for its construction, so we don't need a publication protection, what I mean is if class DefaultRestierDeserializerProvider itself is thread safe? All its members including inherited ones? I've just checked, it seems the provider was used as singleton (for a controller) before this change, so now I think it is thread safe, and a static member is enough I believe.